tensorboard/components/tf_backend/router.ts

```diff
@@ -120,8 +126,8 @@ function createDemoPath(pathPrefix: string, path: string,
   const url = new URL(window.location.origin);
 
   // All demo data are serialized in JSON format.
-  url.pathname = `${pathPrefix}/${normalizedPath}${ext}`;
-  return url.pathname + url.search;
+  url.pathname = `${dataDir}/${normalizedPath}${ext}`;
+  return url.pathname;
```

Comment:
> For instance, the current code will always return a string that starts
with a slash, even if dataDir does not. Is this intentional?

Yup, this is intentional. demos data shims are served at /data/ regardless of --path_prefix so we want to form an absolute path.
About using getter-setter, I was implicitly using URL's ability to sanitize the path and form a legal URL. Because we had undone that part in prodURL, technically, there is no reason to do that for demo url. (moreover, because we do not render output of the router into source of an anchor tag or in dangerous way, I think the sanitization of URL we form is not strictly required so that's that)

> If the latter, could you please leave a comment explaining why
Will do